packages/cli/src/commands/pin/ls.ts

```diff
@@ -0,0 +1,31 @@
+import { StreamID } from '@ceramicnetwork/streamid'
+
+import { Command } from '../../command'
+import type { CommandFlags } from '../../command'
+
+export default class List extends Command<CommandFlags, { streamId?: string }> {
+  static description = 'list pinned streams'
+
+  static args = [
+    {
+      name: 'streamId',
+      description: 'optional stream ID filter',
+    },
+  ]
+
+  async run(): Promise<void> {
+    this.spinner.start('Loading pins list...')
+    try {
+      const id = this.args.streamId ? StreamID.fromString(this.args.streamId) : undefined
+      const iterator = await this.ceramic.pin.ls(id)
+      const pinnedStreamIds = []
+      for await (const id of iterator) {
```

Comment:
this list could be very large, might be better to log them as you iterate over it, rather than trying to pull it all into memory at once.

Reply:
Yes good idea 👍